components/video/VideoPlayerView.bs

```diff
@@ -753,6 +758,9 @@ end sub
 
 sub displaySegmentSkipButton(segmentType as string, endticks as double)
     m.currentSegmentEndTicks = endticks
+    m.currentSegmentType = segmentType
+
+    if isValidAndNotEmpty(m.mediaSegmentDismissed) and m.mediaSegmentDismissed.DoesExist(m.currentSegmentType) and m.mediaSegmentDismissed[m.currentSegmentType] then return
```

Comment:
```suggestion
    if isValidAndNotEmpty(m.mediaSegmentDismissed) and m.mediaSegmentDismissed.DoesExist(m.currentSegmentType) then return
```

Since we're never setting the value to anything but true, checking if the segmentType exists in the dismissed assocarray is sufficient.